scripts/convert-translations.js

```diff
@@ -0,0 +1,212 @@
+        if (!whitelist || whitelist.includes(k)) {
+          const escapedValue = v.replace("'", "\\'");
+          r += `${indent2}${k}: '${escapedValue}',\n`;
+        }
+      } else if (typeof v === 'object') {
+        r = printJsonLevel(v, r, lvl + 1);
+      }
+    });
+
+    r += `${indent}}\n`;
+
+    return r;
+  }
+
+  let result = '';
+
+  // 1. Make it compatible with our linting rules
+  // console.log(fileContent);
+  const parsed = JSON.parse(fileContent);
+  result += printJsonLevel(parsed);
+
+  // 2. Add 'export default '
+  result = `export default ${result}`;
+
+  return result;
+}
+
+function getDerivedLocaleFiles(isoLang) {
+  const /** @type {Array} */ result = [];
+  const /** @type {Array} */ localesForLang = localesSupported[isoLang];
+
+  if (!localesForLang) return undefined;
+
+  localesForLang.forEach(locale => {
+    const derivedFile = `import ${isoLang} from './${isoLang}.js';
+
+export default {
+  ...${isoLang},
+};
+`;
+
+    result.push({ derivedFileName: `${isoLang}-${locale}.js`, derivedContent: derivedFile });
+  });
+
+  return result;
+}
+
+function writeResults(derivedLocaleFiles, filePath, rewrittenContent, { dryRun }) {
+  const results = [];
+  // These are contents of original file
+  results.push({ pathToWrite: filePath.replace('.json', '.js'), newContent: rewrittenContent });
+
+  // These are derived contents
+  const fileFolder = path.dirname(filePath);
+  derivedLocaleFiles.forEach(({ derivedFileName, derivedContent }) => {
+    results.push({ pathToWrite: `${fileFolder}/${derivedFileName}`, newContent: derivedContent });
+  });
+
+  if (dryRun) {
+    results.forEach(({ pathToWrite, newContent }) => {
+      console.log(`== output for filePath '${pathToWrite}' ===`); // eslint-disable-line no-console
+      console.log(newContent); // eslint-disable-line no-console
+    });
+  } else {
+    results.forEach(({ pathToWrite, newContent }) => {
+      fs.writeFileSync(pathToWrite, newContent);
+    });
+  }
+}
+
+/**
+ * Main code
+ */
+function main({ dryRun, cleanup } = { dryRun: false, cleanup: false }) {
+  const jsonFilePaths = gatherFilesFromDir(config.path);
+  jsonFilePaths.forEach(filePath => {
+    // 1. Get contents
+    const content = fs.readFileSync(filePath).toString();
+    const rewrittenContent = rewriteTranslationFile(content, filePath);
+
+    // 2. Add all dialects/locales supported on top
+    const fileName = filePath.replace(/^.*[\\\/]/, ''); // eslint-disable-line
+    const isoLang = fileName.replace('.json', '');
+    const derivedLocaleFiles = getDerivedLocaleFiles(isoLang);
+    if (!derivedLocaleFiles) return;
+
+    // 3. Write the results
+    writeResults(derivedLocaleFiles, filePath, rewrittenContent, { dryRun });
+
+    // 4 Cleanup
+    if (cleanup && !dryRun) {
+      fs.unlinkSync(filePath);
+    }
+  });
+}
+
+module.exports = main;
+
+main({ dryRun: config.dry, cleanup: config.cleanup });
```

Comment:
lion repo is not the best place for this script

please move it somewhere else so that everyone can reuse it, and not only developers within this repository

Reply:
yep, already discussed. This goes 'one layer up' :)